packages/wallets/src/evm/wallet.ts

```diff
@@ -26,48 +68,318 @@ export interface ViemWallet {
+        const signature = await this.approveSignature(signatureId, pendingApproval.message as Hex);
+        if (signature === undefined) {
+            throw new Error("Signature not available");
+        }
+
+        // Get signature status until success
+        let signatureResponse: GetSignatureResponse | null = null;
+        while (signatureResponse === null || signatureResponse.status === "pending") {
+            await sleep(STATUS_POLLING_INTERVAL_MS);
+            signatureResponse = await this.apiClient.getSignature(this.walletLocator, signatureId);
+        }
+
+        if (signatureResponse.status === "failed") {
+            throw new Error("Typed data signing failed");
+        }
+
+        return signature;
     }
 
-    // biome-ignore lint/suspicious/useAwait: <explanation>
     public async sendTransaction(parameters: {
         to: Address;
         data?: Hex;
         value?: bigint;
     }): Promise<Hex> {
-        throw new Error("Not implemented");
+        // Create transaction
+        const transactionCreationResponse = await this.apiClient.createTransaction(this.walletLocator, {
+            params: {
+                signer: this.signerLocator,
+                chain: this.chain,
+                calls: [
+                    {
+                        to: parameters.to,
+                        value: parameters.value ? parameters.value.toString() : "0",
+                        data: parameters.data ?? "0x",
+                    },
+                ],
+            },
+        });
+        const transactionId = transactionCreationResponse.id;
+
+        // Approve transaction
+        const pendingApprovals = transactionCreationResponse.approvals?.pending || [];
+        if (pendingApprovals.length !== 1) {
+            throw new Error(`Expected 1 pending approval, got ${pendingApprovals.length}`);
+        }
+        const pendingApproval = pendingApprovals[0];
+        await this.approveTransaction(transactionId, pendingApproval.message as Hex);
+
+        // Get transaction status until success
+        let transactionResponse: GetTransactionResponse | null = null;
+        while (transactionResponse === null || transactionResponse.status === "pending") {
+            await sleep(STATUS_POLLING_INTERVAL_MS);
+            transactionResponse = await this.apiClient.getTransaction(this.walletLocator, transactionId);
+        }
+
+        if (transactionResponse.status === "failed") {
+            throw new Error("Transaction sending failed");
+        }
+
+        // Get transaction hash
+        const transactionHash = transactionResponse.onChain.txId;
+        if (transactionHash === undefined) {
+            throw new Error("Transaction hash not found");
+        }
+        return transactionHash as Hex;
+    }
+
+    private get walletLocator(): string {
+        if (this.apiClient.isServerSide) {
+            return this.address;
+        } else {
+            return "evm-smart-wallet";
```

Comment:
in client side we're actually returning the walletType, the locator would be me:evm-smart-wallet. I would consider changing the response here and remove the `me` prefix from the apiClient